target/linux/ramips/dts/mt7620a_dlink_dwr-961-a1.dts

```diff
@@ -0,0 +1,241 @@
+		};
+
+		led-2 {
+			function = LED_FUNCTION_LAN;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio0 12 GPIO_ACTIVE_LOW>;
+		};
+
+		led-3 { /* hidden next to wlan5g led */
+			function = LED_FUNCTION_DEBUG;
+			function-enumerator = <1>;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio0 13 GPIO_ACTIVE_LOW>;
+		};
+
+		led-4 { /* hidden next to sms led*/
+			function = LED_FUNCTION_DEBUG;
+			function-enumerator = <2>;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+		};
+
+		led-5 {
+			function = LED_FUNCTION_SMS;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio1 15 GPIO_ACTIVE_LOW>;
+		};
+
+		led-6 {
+			function = LED_FUNCTION_SIGNAL;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio2 0 GPIO_ACTIVE_HIGH>;
+		};
+
+		led-7 {
+			function = LED_FUNCTION_SIGNAL;
+			color = <LED_COLOR_ID_RED>;
+			gpios = <&gpio2 1 GPIO_ACTIVE_HIGH>;
+		};
+
+		led-8 {
+			function = LED_FUNCTION_CONNECTION;
+			function-enumerator = <4>;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio2 2 GPIO_ACTIVE_LOW>;
+		};
+
+		led-9 {
+			function = LED_FUNCTION_CONNECTION;
+			function-enumerator = <3>;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio2 3 GPIO_ACTIVE_LOW>;
+		};
+
+		led-10 {
+			function = LED_FUNCTION_WLAN;
+			function-enumerator = <5>;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio2 4 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy0tpt";
+		};
+
+		led-11 {
+			function = LED_FUNCTION_WLAN;
+			function-enumerator = <2>;
+			color = <LED_COLOR_ID_GREEN>;
+			gpios = <&gpio3 0 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "phy1tpt";
+		};
+	};
+};
+
+&ehci {
+	status = "okay";
+};
+
+&ethernet {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rgmii1_pins &rgmii2_pins &mdio_pins>;
+
+	port@5 {
+		status = "okay";
+		phy-mode = "rgmii";
+		mediatek,fixed-link = <1000 1 1 1>;
+	};
+
+	mdio-bus {
+		status = "okay";
+
+		ethernet-phy@0 {
+			reg = <0>;
+			phy-mode = "rgmii";
+			qca,ar8327-initvals = <
+				0x04 0x87300000 /* PORT0 PAD MODE CTRL */
+				0x0c 0x00000000 /* PORT6 PAD MODE CTRL */
+				0x7c 0x0000007e /* PORT0_STATUS */
+				0x80 0x00001200 /* PORT1_STATUS */
+				0x84 0x00001200 /* PORT2_STATUS */
+				0x88 0x00001200 /* PORT3_STATUS */
+				0x8c 0x00001200 /* PORT4_STATUS */
+				0x90 0x00001200 /* PORT5_STATUS */
+				0x94 0x00000000 /* PORT6_STATUS */
+			>;
+		};
+	};
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&gpio2 {
+	status = "okay";
+};
+
+&gpio3 {
+	status = "okay";
+};
+
+&gsw {
+	mediatek,ephy-base = /bits/ 8 <8>;
+};
+
+&ohci {
+	status = "okay";
+};
+
+&pcie {
+	status = "okay";
+};
+
+&pcie0 {
+	wifi@0,0 {
+		reg = <0x0000 0 0 0 0>;
+		ieee80211-freq-limit = <5000000 6000000>;
+		nvmem-cells = <&macaddr_config_e50e>;
+		nvmem-cell-names = "mac-address";
+		mac-address-increment = <(2)>;
+		mediatek,mtd-eeprom = <&config 0xe29e>;
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
```

Comment:
Can it go faster?

Reply:
JBOOT devices are known from extremely slow spi-nor. I'm afraid to go faster.

Reply:
Okay, but could you elaborate? You added c948a4782ba9 with 50 MHz, but you think it was too high?

Reply:
First known devices with jboot (designed by Amit)  have flash, which work well only on low speed.
I copied speed from other device, but here spi is similar to dwr-960, and work well. So I switch to 50M.

```
root@OpenWrt:~# time dd if=/dev/mtd1 of=/dev/null bs=128k # 50M
127+0 records in
127+0 records out
real	0m 9.00s
user	0m 0.00s
sys	0m 8.94s
root@OpenWrt:~#
time dd if=/dev/mtd1 of=/dev/null bs=128k #10M
127+0 records in
127+0 records out
real	0m 31.01s
user	0m 0.00s
sys	0m 30.91s
```